azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -0,0 +1,360 @@
+                )
+            else:
+                raise ArgumentUsageError(
+                    error_msg + " or endpoint account '--endpoint-account'."
+                )
+
+        endpoints = self.hub_resource.properties.routing.endpoints
+        if EndpointType.EventHub.value == endpoint_type.lower():
+            if authentication_type != AuthenticationType.IdentityBased.value and not connection_string:
+                connection_string = get_eventhub_cstring(
+                    cmd=self.cmd,
+                    namespace_name=endpoint_account_name,
+                    eventhub_name=entity_path,
+                    policy_name=endpoint_policy_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            endpoints.event_hubs.append({
+                "connectionString": connection_string,
+                "name": endpoint_name,
+                "subscriptionId": endpoint_subscription_id,
+                "resourceGroup": endpoint_resource_group,
+                "authenticationType": authentication_type,
+                "endpointUri": endpoint_uri,
+                "entityPath": entity_path,
+                "identity": endpoint_identity
+            })
+        elif EndpointType.ServiceBusQueue.value == endpoint_type.lower():
+            if authentication_type != AuthenticationType.IdentityBased.value and not connection_string:
+                connection_string = get_servicebus_queue_cstring(
+                    cmd=self.cmd,
+                    namespace_name=endpoint_account_name,
+                    queue_name=entity_path,
+                    policy_name=endpoint_policy_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            endpoints.service_bus_queues.append({
+                "connectionString": connection_string,
+                "name": endpoint_name,
+                "subscriptionId": endpoint_subscription_id,
+                "resourceGroup": endpoint_resource_group,
+                "authenticationType": authentication_type,
+                "endpointUri": endpoint_uri,
+                "entityPath": entity_path,
+                "identity": endpoint_identity
+            })
+        elif EndpointType.ServiceBusTopic.value == endpoint_type.lower():
+            if authentication_type != AuthenticationType.IdentityBased.value and not connection_string:
+                connection_string = get_servicebus_topic_cstring(
+                    cmd=self.cmd,
+                    namespace_name=endpoint_account_name,
+                    topic_name=entity_path,
+                    policy_name=endpoint_policy_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            endpoints.service_bus_topics.append({
+                "connection_string": connection_string,
```

Comment:
Just to clarify, service bus queues use `connectionString` - does topic use `connection_string` or does this need changed?

Reply:
good q - both prob work